improved telemetry documentation (#15415)
kyma-project/kyma: docs/01-overview/main-areas/observability/obsv-04-telemetry-in-kyma.md

Kyma's observability functionality focuses on the pre-integrated collection of telemetry data from the users' workloads, and making that data available for further analysis and storage in backends of any kind and location. The telemetry component provides the configurable collection and shipment components, and with that, separates them explicitly from the storage and analysis in a specific backend system. You can still choose to install lightweight in-cluster backends with dedicated observability components.

> To support the logging domain, the telemetry component provides a log collector, Fluent Bit. You can configure the log collector with external systems using runtime configuration with a dedicated Kubernetes API (CRD) named `LogPipeline`. With that, you can integrate with vendors natively such as [VMWare](https://medium.com/@shrishs/log-forwarding-from-fluent-bit-to-vrealizeloginsightcloud-9eeb14b40276) using the `http` output, or with any vendor via a [fluentd integration](https://medium.com/hepsiburadatech/fluent-logging-architecture-fluent-bit-fluentd-elasticsearch-ca4a898e28aa) using the as well the `http` output. Furthermore, you can run the collector in an [unsupported mode](#unsupported-mode), leveraging the full output options of FluentBit. Additionally, you can at any time bring your own log collector if advanced configuration options are required.

Reviewer: ```suggestion
To support the logging domain, the Telemetry component provides a log collector, Fluent Bit. You can configure the log collector with external systems using runtime configuration with a dedicated Kubernetes API (CRD) named `LogPipeline`. Using the LogPipeline's HTTP output, you can natively integrate with vendors such as [VMWare](https://medium.com/@shrishs/log-forwarding-from-fluent-bit-to-vrealizeloginsightcloud-9eeb14b40276), or with any vendor using a [Fluentd integration](https://medium.com/hepsiburadatech/fluent-logging-architecture-fluent-bit-fluentd-elasticsearch-ca4a898e28aa). Furthermore, you can run the collector in an [unsupported mode](#unsupported-mode), leveraging the full output options of Fluent Bit. Additionally, you can always bring your own log collector if you need advanced configuration options.
```

3. Fluent Bit queries the [Kubernetes API Server](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/) for additional Pod metadata, such as Pod annotations and labels.

> 4. The telemetry component configures Fluent Bit with your custom output configuration.

Reviewer: ```suggestion
4. The Telemetry component configures Fluent Bit with your custom output configuration.
```

5. If Kyma's logging component is installed, the operator configures the shipment to the in-cluster Loki instance automatically.

> 6. As specified in your `LogPipeline` configuration, Fluent Bit sends the log data to observability systems outside or inside the Kyma cluster. Here, mainly the integration via `HTTP` is available which can be used to integrate a system directly or via an additional FluentD installation.

Reviewer: ```suggestion
6. As specified in your `LogPipeline` configuration, Fluent Bit sends the log data to observability systems outside or inside the Kyma cluster. Here, mainly the integration with HTTP is available, with which you can integrate a system directly or with an additional Fluentd installation.
```

What does "mainly" mean here? 90% of the cases?

    ```
    An output is a data destination configured by a [Fluent Bit output](https://docs.fluentbit.io/manual/pipeline/outputs) of the relevant type. The LogPipeline supports the following output types:

>     - **http**, which sends the data to the specified http destination. The output is designed to integrate with an [FluentD HTTP Input](https://docs.fluentd.org/input/http) which opens up a huge ecosystem of integration possibilities.

Reviewer: ```suggestion
    - **http**, which sends the data to the specified HTTP destination. The output is designed to integrate with a [Fluentd HTTP Input](https://docs.fluentd.org/input/http), which opens up a huge ecosystem of integration possibilities.
```

    - **grafana-loki**, which sends the data to the Kyma-internal Loki instance. Note: this output is considered legacy and is only available for backwards compatibility with the in-cluster Loki instance. It might not be compatible with latest Loki versions. For integration with Loki, use the `custom` output with name `loki` instead.

>     - **custom**, which supports the configuration of any destination in the Fluent Bit configuration syntax. Note: Usage of a `custom` output will put the LogPipeline in [unsupported mode](#unsupported-mode).

Reviewer: I suggest to put this as the 2nd bullet and `grafana-loki` as 3rd, considering a) `grafana-loki` mentions `custom`, so it `custom` should be introduced earlier, and b) `grafana-loki` is legacy and thus should come last.

```suggestion
    - **custom**, which supports the configuration of any destination in the Fluent Bit configuration syntax. Note: If you use a `custom` output, you put the LogPipeline in [unsupported mode](#unsupported-mode).
```

| output.http.user | object | Basic Auth username.|
| output.http.user.value | string | Username value, can contain references to Secret values. |
| output.http.user.valueFrom.secretKeyRef | object | Reference to a key in a Secret. You must provide `name` and `namespace` of the Secret, as well as the name of the `key`. |

> | output.custom | string | Any other Fluent Bit output specified in the Fluent Bit configuration syntax. Note: Usage of a `custom` output will put the LogPipeline in [unsupported mode](#unsupported-mode). |

Reviewer: ```suggestion
| output.custom | string | Any other Fluent Bit output specified in the Fluent Bit configuration syntax. Note: If you use a `custom` output, you put the LogPipeline in [unsupported mode](#unsupported-mode). |
```

### Unsupported Mode

> The `unsupportedMode` attribute of a `LogPipeline` indicates that you are using a `custom` filter and/or output. The Kyma Team is not providing support for a custom configuration.

Reviewer: ```suggestion
The `unsupportedMode` attribute of a `LogPipeline` indicates that you are using a `custom` filter and/or `custom` output. The Kyma team does not provide support for a custom configuration.
```
